torch_xla_py/xla_dist.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+"""Tool to distribute training on Cloud TPU Pods."""
+from __future__ import division
+from __future__ import print_function
+
+import abc
+
+class BaseWorker(abc.ABC):
+
+  @classmethod
+  @abc.abstractmethod
+  def validate_cluster(cls, workers):
+    """Validates the current worker cluster configuration.
+
+    Raises:
+      RuntimeError: If the cluster is misconfigured, this validation will
+      raise a error.
+    """
+    raise NotImplementedError()
+
+
+class CpuWorker(BaseWorker):
```

Comment:
I am not sure `CpuWorker` is correct here.
Maybe `ClientWorker` and `ServiceWorker` to match what we call the meshes?
But then with 1VM client/service might go away.
In the end, I think Client/Service are maybe a better match than Cpu/Tpu.

Open to other ideas ...

Reply:
Done.

Reply:
Actually I've updated it such that we have no distinction between `Client` and `Service` clusters. Currently, I don't see enough differences between the two to warrant that.